flytekit/core/type_engine.py

```diff
@@ -167,12 +167,16 @@ def isinstance_generic(self, obj, generic_alias):
 
         # Optionally check the type of elements if it's a collection like list or dict
         if origin in {list, tuple, set}:
+            if len(obj) == 0:
+                return
             for item in obj:
                 self.assert_type(args[0], item)
                 return
             raise TypeTransformerFailedError(f"Not all items in '{obj}' are of type {args[0]}")
```

Comment:
it's also not currently the case that this block asserts that _every_ `item` in `obj` is of type `args[0]`, rather just that the first item is of the correct type

Reply:
But someone could add two items with different types to the list, right?

Reply:
Yes, I left the bulk of the logic unchanged. If you like I can fix this to do what it seems like it was trying to do…

Reply:
updated to fix the logic to check every item.